tests/base.py

```diff
@@ -0,0 +1,305 @@
+                self.PRIMARY_KEYS: {"webinar_id"},
+                self.REPLICATION_METHOD: self.FULL_TABLE,
+                self.RESPECTS_START_DATE: False,
+            },
+            'webinar_tracking_sources': {
+                self.PRIMARY_KEYS: {"webinar_id", "id"},
+                self.REPLICATION_METHOD: self.FULL_TABLE,
+                self.RESPECTS_START_DATE: False,
+            },
+            'report_webinars': {
+                self.PRIMARY_KEYS: {"uuid"},
+                self.REPLICATION_METHOD: self.FULL_TABLE,
+                self.RESPECTS_START_DATE: False,
+            },
+            'report_webinar_participants': {
+                self.PRIMARY_KEYS: {"webinar_id", "id"},
+                self.REPLICATION_METHOD: self.FULL_TABLE,
+                self.RESPECTS_START_DATE: False,
+            },
+        }
+
+    def expected_automatic_fields(self):
+        auto_fields = {}
+        for k, v in self.expected_metadata().items():
+            auto_fields[k] = v.get(self.PRIMARY_KEYS, set()) | v.get(self.REPLICATION_KEYS, set())
+
+        return auto_fields
+
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass(logging="Ensuring environment variables are sourced.")
+        missing_envs = [
+            x for x in [
+                "TAP_ZOOM_CLIENT_ID", "TAP_ZOOM_CLIENT_SECRET", "TAP_ZOOM_REFRESH_TOKEN"
+            ] if os.getenv(x) is None
+        ]
+
+        if len(missing_envs) != 0:
+            raise Exception("Missing environment variables: {}".format(missing_envs))
+
+
+    ##########################################################################
+    ### Tap Specific Methods
+    ##########################################################################
+
+
+    @staticmethod
+    def expected_pagination_fields(): # TODO does this apply?
+        return {
+            "Test Report 1" : set(),
+            "Audience Overview": {
+                "ga:users", "ga:newUsers", "ga:sessions", "ga:sessionsPerUser", "ga:pageviews",
+                "ga:pageviewsPerSession", "ga:sessionDuration", "ga:bounceRate", "ga:date",
+                # "ga:pageviews",
+            },
+            "Audience Geo Location": set(),
+            "Audience Technology": set(),
+            "Acquisition Overview": set(),
+            "Behavior Overview": set(),
+            "Ecommerce Overview": set(),
+        }
+
+
+
+    # TODO refactor code to remove this method if possible, ga4 relic
+    def get_stream_name(self, stream):
+        """
+        Returns the stream_name given the tap_stream_id because synced_records
+        from the target output batches records by stream_name
+
+        Since the GA4 tap_stream_id is a UUID instead of the usual case of
+        tap_stream_id == stream_name, we need to get the stream_name that
+        maps to tap_stream_id
+        """
+        stream_name=stream
+        # custom_reports_names_to_ids().get(tap_stream_id, tap_stream_id)
+        return stream_name 
```

Comment:
This is copied from GA4 and the streams, fields, and comment blocks all reference GA4 specific information. I assume these will need to be removed or cleaned up in order to be relevant

Reply:
Done.